test/e2e/framework.go

```diff
@@ -611,6 +617,138 @@ func validatePodIP(podNetworkCIDR, podIP string) (bool, error) {
 	return cidr.Contains(ip), nil
 }
 
+// createNginxPodOnNode creates a nginx pod listening for traffic.
+func (data *TestData) createNginxPodOnNode(podName string, nodeName string) error {
+	podSpec := v1.PodSpec{
+		Containers: []v1.Container{
+			{
+				Name:            defaultNginxContainerName,
+				Image:           "nginx",
+				ImagePullPolicy: v1.PullIfNotPresent,
+			},
+		},
+		RestartPolicy: v1.RestartPolicyNever,
+	}
+	if nodeName != "" {
+		podSpec.NodeSelector = map[string]string{
+			"kubernetes.io/hostname": nodeName,
+		}
+	}
+	if nodeName == masterNodeName() {
+		// tolerate NoSchedule taint if we want Pod to run on master node
+		noScheduleToleration := v1.Toleration{
+			Key:      "node-role.kubernetes.io/master",
+			Operator: v1.TolerationOpExists,
+			Effect:   v1.TaintEffectNoSchedule,
+		}
+		podSpec.Tolerations = []v1.Toleration{noScheduleToleration}
+	}
+	pod := &v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: podName,
+			Labels: map[string]string{
+				"pod-name": podName,
+				"app":      "nginx",
+			},
+		},
+		Spec: podSpec,
+	}
+	if _, err := data.clientset.CoreV1().Pods(testNamespace).Create(pod); err != nil {
+		return err
+	}
+	return nil
+}
+
+// createNginxService exposes nginx as a service.
+func (data *TestData) createNginxService(podName string, svcName string) error {
+	port := 80
+	_, err := data.clientset.CoreV1().Services(testNamespace).Create(&v1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: svcName,
+			Labels: map[string]string{
+				"app": "nginx",
+			},
+		},
+		Spec: v1.ServiceSpec{
+			Ports: []v1.ServicePort{{
+				Port:       int32(port),
+				TargetPort: intstr.IntOrString{Type: 0, IntVal: int32(port)},
+			}},
+			Selector: map[string]string{
+				"app": "nginx",
+			},
+		},
+	})
+	return err
+}
+
+// deleteNginxService deletes the nginx service
+func (data *TestData) deleteNginxService(svcName string) error {
+	if err := data.clientset.CoreV1().Services(testNamespace).Delete(svcName, nil); err != nil {
+		return fmt.Errorf("unable to cleanup svc %v: %v", svcName, err)
+	}
+	return nil
+}
+
+// createNetworkPolicyToTestIPBlockWithExcept creates a network policy with IPBlock's Except .
+// 192.168.1.2 ~ 192.168.1.255 allowd
+// 192.168.1.0 ~ 192.168.1.1 rejected.
```

Comment:
Are they related?

Reply:
Removed. I thought an example is needed but it seems unnecessary...